flask-server/main.py

```diff
@@ -526,11 +500,118 @@ def unshare_clink():
+            bookmark['link'] = link
+        if title:
+            bookmark['title'] = title
+        if description: 
+            bookmark['description'] = description
+
+        datastore_client.put(bookmark)
+
+        return bookmark_entity_to_return(bookmark), 200
+    else: 
+        response_object = {
+            'status': 'fail',
+            'message': 'Invalid Authorization. Failed to edit bookmark.'
+        }
+        return response_object, 401
+
+# edit clink api
+@app.route('/apis/edit-clink', methods=['POST'])
+def edit_clink():
+    resp_token = decode_auth_token(request.json['Authorization'])
+    clink_id = request.json['clinkId']
+
+    if is_valid_instance(resp_token) and has_write_access(clink_id, resp_token):
+        title = request.json['title']
+
+        key = datastore_client.key('clink', int(clink_id))
+        clink = datastore_client.get(key)
+        
+        if title:
+            clink['title'] = title
+
+            datastore_client.put(clink)
+
+        return clink_entity_to_return(clink), 200
+    else: 
+        response_object = {
+            'status': 'fail',
+            'message': 'Invalid Authorization. Failed to edit clink.'
+        }
+        return response_object, 401
+
+# delete bookmark api
+@app.route('/apis/delete-bookmark', methods=['POST'])
+def delete_bookmark():
+    resp_token = decode_auth_token(request.json['Authorization'])
+    clink_id = request.json['clinkId']
+
+    if is_valid_instance(resp_token) and has_write_access(clink_id, resp_token):
+
+        key = datastore_client.key('bookmark', int(request.json['bookmarkId']))
+        bookmark = datastore_client.get(key)
+        
+        bookmark['deleted'] = True
+
+        datastore_client.put(bookmark)
+
+        return bookmark_entity_to_return(bookmark), 200
+    else: 
+        response_object = {
+            'status': 'fail',
+            'message': 'Invalid Authorization. Failed to edit bookmark.'
+        }
+        return response_object, 401
+
+# delete clink api
+@app.route('/apis/delete-clink', methods=['POST'])
+def delete_clink():
+    resp_token = decode_auth_token(request.json['Authorization'])
+    clink_id = request.json['clinkId']
+
+    if is_valid_instance(resp_token) and has_write_access(clink_id, resp_token):
+
+        key = datastore_client.key('clink', int(clink_id))
+        clink = datastore_client.get(key)
+        
+        clink['deleted'] = True
+
+        datastore_client.put(clink)
+
+        return clink_entity_to_return(clink), 200
+    else: 
+        response_object = {
+            'status': 'fail',
+            'message': 'Invalid Authorization. Failed to edit clink.'
+        }
+        return response_object, 401
+
+# write access api
+def has_write_access(clink_id, user_id):
+    access = True if clink_id == "All" else list(datastore_client.query(kind='user_write_map').add_filter('clink_id', '=', int(clink_id)).add_filter('user_id', '=', int(user_id)).fetch(limit=1))[0]
```

Comment:
can this be
```
access = clink_id == "All" or list(datastore_client.query(kind='user_write_map').add_filter('clink_id', '=', int(clink_id)).add_filter('user_id', '=', int(user_id)).fetch(limit=1))[0]
return bool(access)
```
Alternatively if you're only using has_write_access as a bool, then not casting it as bool(access) should probably still work as it'll return an entity but you're using the entity in an if statement anyway